.ci/gitlab/test.yml

```
@@ -45,40 +63,38 @@ build:
 
     # Archive all build files (from .cabal and dist-newstyle)
     - tar -cf - $(.ci/get_build_dist.sh) | zstd -T${THREADS} -15 > dist.tar.zst
-  tags:
-   - local
 
 # Tests run on shared runners:
 
 cores:unittests:
-  extends: .test-common
+  extends: .test-nocache
   script:
-    - cabal v2-run -- clash-cores:unittests --hide-successes
+    - ./dist-newstyle/build/*/*/clash-cores-*/t/unittests/build/unittests/unittests --hide-successes
```

```suggestion
    - ./.ci/run_test.sh clash-cores unittests --hide-successes
```

Moving all the `*` stuff to a separate script seems a little bit cleaner to me.

---

That will need a bit more than that as there are three patterns:
1. `./dist-newstyle/build/*/*/clash-prelude-*/t/unittests/build/unittests/unittests`
2.`./dist-newstyle/build/*/*/clash-cosim-*/build/test/test`
3. `/dist-newstyle/build/*/*/clash-testsuite-*/x/clash-testsuite/build/clash-testsuite/clash-testsuite`

What's the difference between 1 and 2, you reckon? Whether there are multiple `test-suite` stanzas in the `.cabal` file or not? As in, `clash-cosim` has only one, maybe that's why it has a different path to the executable. We could also keep that one separate and make a script for 1 and 3 only.

---

`clash-cosim` has a custom setup, perhaps that's it?

> The fact that I couldn't readily tell what the structure was, was the reason to forego the separate script solution initially. 

Sure, that makes sense.

> Do you want to make this nicer

Nah it's fine. I'd probably have done something like `find dist-newstyle/ -executable -type f -name $bin_name | grep $lib_name`, but then again, that doesn't feel particularly nice either.